chia/consensus/blockchain.py

```diff
@@ -806,27 +806,14 @@ async def pre_validate_blocks_multiprocessing(
         self,
         blocks: List[FullBlock],
         npc_results: Dict[uint32, NPCResult],  # A cache of the result of running CLVM, optional (you can use {})
+        sub_slot_iters: uint64,
+        difficulty: uint64,
+        prev_ses_block: Optional[BlockRecord],
```

Comment:
I really think it would make things simpler to have a class that lumps these 3 fields together. `pre_validate_blocks_multiprocessing()` would take this class as a single parameter instead of 3 separate ones.

Maybe it could even have a constructor function that computes the values.

Reply:
a constructor will not help since there is no one way of getting these values it depends on the context 

personally i dont think a class would help us here since its only used in calls to pre_validate_blocks_multiprocessing

Reply:
You also pass it to `add_block_batch()`. In the future I intend to split up `add_block_batch()` into smaller pieces, to allow pipelining them. Those smaller pieces will also need these to be passed (and returned).

I think the main benefit, however, is that you can pass a class as an in-out parameter, rather than returning 3 additional values. But it might make sense to punt on this until the need is greater.